### PR TITLE
[UX] Improve error handling for Meshery-Server-1085 on send design failure

### LINKED ISSUE #12233: [UX] Meshery Server:  Error code - Meshery-Server-1085 needs much better cause and remediations.
### Current Behavior
"413 Fall to send design to remote provider." Doesn’t make sense to users. 

1. Use the name of the remote provider.
2. State that design could be corrupt; to undo changes.

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by shinigami-777**
**Notes for Reviewers**

This PR fixes #12233 

- Add clearer causes and remediation for Error Meshery-Server-1085.
- Handle 413 (Payload Too Large), 401 (Unauthorized), and 400 (Bad Request) status codes with specific error messages.
- Include remote provider name in error logs for better clarity and context.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Thanks for contributing, @shinigami-777!

**Comment by vishalvivekm** [comment]
@shinigami-777 
Thank you for your contributions! Let's discuss this during the Meshery Development call today at 9:00 AM CT. Adding it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#) || [meeting link](https://meet.layer5.io/meshery).

**Review by alphaX86 — COMMENTED** [review]
Thanks a lot for your PR @shinigami-777 

Has this been tested in local? Just want to confirm before proceeding

**Comment by shinigami-777** [comment]
@alphaX86 Yes it was tested.

**Comment by alphaX86** [comment]
> @alphaX86 Yes it was tested.

Okay...

File: server/models/remote_provider.go [hunk 1]
```diff
@@ -1665,9 +1665,25 @@ func (l *RemoteProvider) SaveMesheryPattern(tokenString string, pattern *Meshery
...
+		case http.StatusUnauthorized:
+			err = ErrPost(fmt.Errorf("failed to send design %s to remote provider %s: Unauthorized access. Check your credentials.", pattern.Name, l.ProviderName), "", resp.StatusCode)
+			l.Log.Error(err)
+			return bdr, err
+		case http.StatusBadRequest:
+			err = ErrPost(fmt.Errorf("failed to send design %s to remote provider %s: Bad request. The design might be corrupt.", pattern.Name, l.ProviderName), "", resp.StatusCode)
+			l.Log.Error(err)
+			return bdr, err
+		default:
+			err = ErrPost(fmt.Errorf("failed to send design %s to remote provider %s. Check if the design is valid or undo recent changes.", pattern.Name, l.ProviderName), "", resp.StatusCode)
```
Comment (alphaX86) [inline server/models/remote_provider.go:1683]: This seems fine to me.

**Review by alphaX86 — CHANGES_REQUESTED** [review]

**Review by alphaX86 — APPROVED** [review]
LGTM. Approved from my side

**Review by lekaf974 — APPROVED** [review]

File: server/models/remote_provider.go [hunk 2]
```diff
@@ -1667,9 +1667,25 @@ func (l *RemoteProvider) SaveMesheryPattern(tokenString string, pattern *Meshery
...
+			l.Log.Error(err)
+			return bdr, err
```
Comment (Jougan-0) [inline server/models/remote_provider.go:1687]: I think if we are logging it here we don't need to return back the error.

File: server/models/remote_provider.go [hunk 3]
```diff
@@ -1667,9 +1667,25 @@ func (l *RemoteProvider) SaveMesheryPattern(tokenString string, pattern *Meshery
...
+		case http.StatusUnauthorized:
+			err = ErrPost(fmt.Errorf("failed to send design %s to remote provider %s: Unauthorized access. Check your credentials.", pattern.Name, l.ProviderName), "", resp.StatusCode)
+			l.Log.Error(err)
+			return bdr, err
+		case http.StatusBadRequest:
+			err = ErrPost(fmt.Errorf("failed to send design %s to remote provider %s: Bad request. The design might be corrupt.", pattern.Name, l.ProviderName), "", resp.StatusCode)
+			l.Log.Error(err)
+			return bdr, err
```
Comment (Jougan-0) [inline server/models/remote_provider.go:1683]: Same here.

File: server/models/remote_provider.go [hunk 4]
```diff
@@ -1667,9 +1667,25 @@ func (l *RemoteProvider) SaveMesheryPattern(tokenString string, pattern *Meshery
...
+
+	switch resp.StatusCode {
+		case http.StatusRequestEntityTooLarge:
+			err = ErrPost(fmt.Errorf("failed to send design %s to remote provider %s: Design file is too large to upload. Reduce the file size and try again", pattern.Name, l.ProviderName), "", resp.StatusCode)
+			l.Log.Error(err)
+			return bdr, err
+		case http.StatusUnauthorized:
+			err = ErrPost(fmt.Errorf("failed to send design %s to remote provider %s: Unauthorized access. Check your credentials.", pattern.Name, l.ProviderName), "", resp.StatusCode)
+			l.Log.Error(err)
+			return bdr, err
```
Comment (Jougan-0) [inline server/models/remote_provider.go:1679]: Same here.

File: server/models/remote_provider.go [hunk 5]
```diff
@@ -1667,9 +1667,25 @@ func (l *RemoteProvider) SaveMesheryPattern(tokenString string, pattern *Meshery
...
 	}
-	err = ErrPost(fmt.Errorf("failed to send design %s to remote provider", pattern.Name), "", resp.StatusCode)
-	l.Log.Error(err)
-	return bdr, err
+
+	switch resp.StatusCode {
+		case http.StatusRequestEntityTooLarge:
+			err = ErrPost(fmt.Errorf("failed to send design %s to remote provider %s: Design file is too large to upload. Reduce the file size and try again", pattern.Name, l.ProviderName), "", resp.StatusCode)
+			l.Log.Error(err)
+			return bdr, err
```
Comment (Jougan-0) [inline server/models/remote_provider.go:1675]: Same here.

Comment (shinigami-777) [reply server/models/remote_provider.go:1674]: I have removed the error logs for this.

**Review by alphaX86 — APPROVED** [review]

**Review by lekaf974 — APPROVED** [review]
LGTM

**Comment by alphaX86** [comment]
Thanks @shinigami-777 for the PR. Apologies for late delay though in response

**Pull request merged**